src/main/kotlin/no/nav/familie/ef/søknad/service/OppslagServiceLoggHjelper.kt

```diff
@@ -0,0 +1,97 @@
+        if (builder.isNotBlank()) {
+            secureLogger.warn(builder.toString())
+        }
+        return builder.toString()
+    }
+
+    private fun leggTilBarneDiff(søkerinfoV1: Søkerinfo,
+                                 søkerinfoV2: Søkerinfo,
+                                 builder: StringBuilder) {
+        if (søkerinfoV1.barn.size != søkerinfoV2.barn.size) {
+            builder.append("Ikke samme antall barn fra tps og pdl: \n V1: ${søkerinfoV1.barn}, \n V2: ${søkerinfoV2.barn}")
+        } else {
+            val barn1Liste = søkerinfoV1.barn.sortedBy { it.fnr }
+            val barn2Liste = søkerinfoV2.barn.sortedBy { it.fnr }
+            val zip = barn1Liste.zip(barn2Liste)
+            zip.forEach {
+                for (prop in Barn::class.memberProperties) {
+                    val detErDiff = when (prop.name) {
+                        "navn" -> {
+                            listOfSeparatedNames(it.first.navn) != listOfSeparatedNames(it.second.navn)
+                        }
+                        else -> prop.get(it.first) != prop.get(it.second)
+                    }
+                    if (detErDiff) {
+                        builder.append("\n Barn: ${prop.name} = V1: ${prop.get(it.first)}, V2: ${prop.get(it.second)}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun listOfSeparatedNames(name: String) =
+            name.split(" ").sorted()
+
+    private fun leggTilSøkerDiff(søkerinfoV1: Søkerinfo,
+                                 søkerinfoV2: Søkerinfo,
+                                 builder: StringBuilder) {
+        for (prop in Person::class.memberProperties) {
+            val property1 = prop.get(søkerinfoV1.søker)
+            val property2 = prop.get(søkerinfoV2.søker)
+            when (prop.name) {
+                "sivilstand" -> {
+                    logSivilstandsDiff(builder, søkerinfoV1.søker.sivilstand, søkerinfoV2.søker.sivilstand)
+                }
+                else -> {
+                    if (property1 != property2 && prop.name != "sivilstand") {
+                        builder.append("\n Person: ${prop.name} = V1: $property1, V2: $property2")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun logSivilstandsDiff(builder: StringBuilder, sivilstand1: String, sivilstand2: String) {
+
+        val erLikSivilstand = !when (sivilstand1) {
+            "SEPA", "SEPARERT" -> {
+                listOf("SEPA", "SEPARERT").contains(sivilstand2)
+            }
+            "UGIF", "UGIFT" -> {
+                listOf("UGIF", "UGIFT").contains(sivilstand2)
+            }
+            "SKIL", "SKILT" -> {
+                listOf("SKIL", "SKILT").contains(sivilstand2)
+            }
+            "ENKE", "ENKE_ELLER_ENKEMANN" -> {
+                listOf("ENKE", "ENKE_ELLER_ENKEMANN").contains(sivilstand2)
+            }
+            else -> sivilstand1 == sivilstand2
```

Comment:
Tror det kan være bra å få med noen andre typer også?
Er også tilsrekkelig å eks mappe over den fra tps

```
    private fun logSivilstandsDiff(builder: StringBuilder, tpsSivilstand: String, pdlSivilstand: String) {
        val tpsTilPdlSivilstand = when (tpsSivilstand) {
            "REPA" -> "PARTNER"
            "SEPA" -> "SEPARERT"
            "SEPR" -> "SEPARERT_PARTNER"
            "GJPA" -> "GJENLEVENDE_PARTNER"
            "UGIF" -> "UGIFT"
            "SAMB" -> "UGIFT"
            "SKIL" -> "SKILT"
            "ENKE" -> "ENKE_ELLER_ENKEMANN"
            else -> tpsSivilstand
        }
        val erLikSivilstand = tpsTilPdlSivilstand == pdlSivilstand
```



 Øvrige finnes her https://github.com/navikt/familie-ef-soknad/pull/385/files

Reply:
Je hvis vi kunne være sikre på at param 1 er tps og papam to er pdl. Vanligvis ville jeg ikke hatt lyst til å stole på det, men her kunne det kanskje vært greit. Dette er jo midlertidig kode som ikke skal brukes av andre.

Reply:
Men det har vi vel full peiling på? Vi vet ju att vi kaller denne metoden med data fra tps i første argumentet

Reply:
Lager en ny branch med oppdateringer her